packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -797,3 +802,26 @@ const TabListContext = React.createContext<
 >(undefined);
 
 export default Tabs;
+
+// ----------------------------------------------------------------------------
+
+/**
+ * This conditionally adds `scrollbar-gutter: stable` only if the content overflows.
+ * This is a workaround to prevent layout shift that happens because of scrollbar width.
+ *
+ * @see https://github.com/iTwin/iTwinUI/issues/1627
+ */
+const useScrollbarGutter = () => {
```

Comment:
Is it called only once? Or does it listen to height changes of the parent?

If the parent become larger, should we listen for changes and remove the `scrollbar-gutter` when there is no longer an overflow?


https://github.com/iTwin/iTwinUI/assets/45748283/4cb71079-73ca-4926-9d73-87be95d1262e

Reply:
yes it's only called once. I don't think it's worth the extra code for listening to resizes.

the main issue is the layout shift between initial render and subsequent clicks. if the scrollbar-gutter stays after that, it's no big deal